src/functions_math.jl

```diff
@@ -127,3 +132,152 @@ function symmetric_names(L::Tuple{Symbol,Symbol}, dims::Integer)
+            data = AbstractFFTs.$fun(parent(A), $(extra...), numerical_dims)
+            newL = _rename(L, p, ps...)
+            return NamedDimsArray(data, newL)
+        end
+
+        """
+            F = $($plan_fun)(A, :time)
+            A∿ = F * A
+            A ≈ F \\ A∿ ≈ inv(F) * A∿
+
+        FFT plans for `NamedDimsArray`s, identical to `A∿ = $($fun)(A, :time)`.
+        Note you cannot specify the final name, it always transforms `:time => :time∿`.
+        And that the plan `F` stores which dimension number to act on, not which name.
+        """
+        function AbstractFFTs.$plan_fun(A::NamedDimsArray, $(extra...), dims = ntuple(identity, ndims(A)); kw...)
+            dims isa Pair && throw(ArgumentError("$($plan_fun) does not store final names, got Pair $dims"))
+            numerical_dims = Tuple(dim(A, dims))
+            AbstractFFTs.$plan_fun(parent(A), $(extra...), numerical_dims; kw...)
+        end
+    end
+
+end
+
+for shift in (:fftshift, :ifftshift)
+    @eval begin
+
+        function AbstractFFTs.$shift(A::NamedDimsArray)
+            data = AbstractFFTs.$shift(parent(A))
+            NamedDimsArray(data, dimnames(A))
+        end
+
+        function AbstractFFTs.$shift(A::NamedDimsArray, dims)
+            numerical_dims = dim(A, dims)
+            data = AbstractFFTs.$shift(parent(A), numerical_dims)
+            NamedDimsArray(data, dimnames(A))
+        end
+
+    end
+end
+
+# The dimensions on which plans act are not part of the type, unfortunately
+for plan_type in (:Plan, :ScaledPlan)
+    @eval function Base.:*(plan::AbstractFFTs.$plan_type, A::NamedDimsArray{L,T,N}) where {L,T,N}
+        data = plan * parent(A)
+        if Base.sym_in(:region, propertynames(plan)) # true for plan_fft from FFTW
+            dims = plan.region  # dims can be 1, (1,3) or 1:3
+        elseif Base.sym_in(:p, propertynames(plan))
+            dims = plan.p.region
+        else
+            return data
+        end
+        newL = ntuple(d -> d in dims ? wave_name(L[d]) : L[d], N)::NTuple{N,Symbol}
+        # newL = wave_name(L, Tuple(dims)) # this, using compile_time_return_hack, is much slower
+        return NamedDimsArray(data, newL)
+    end
+end
+
+#=
+using NamedDims, FFTW, LinearAlgebra, BenchmarkTools
+
+nda = NamedDimsArray(rand(4,4), (:k, :l))       # nda = NamedDimsArray(rand(ComplexF64, 256, 256), (:k, :l))
+pp = plan_fft(nda)
+
+@btime fft($nda)           # 18.600 μs          # 977.612 μs
+@btime fft($(parent(nda))) # 18.565 μs          # 980.390 μs
+@code_warntype fft(nda)    # looks good
+
+@btime plan_fft($nda, :k)
+@btime plan_fft($nda, (:k,:l)) # 18.393 μs      #    42.622 μs
+
+@btime $pp * $nda             #   9.029 μs      # 955.534 μs
+@btime $pp * $(parent(nda))   #     125.652 ns  # 943.517 μs
+@code_warntype pp * nda       # pretty awful!
+
+ndc = NamedDimsArray(zeros(4,4) .+ im, (:p, :q))
+mul!(ndc.data, pp, nda.data .+ 0im)
+mul!(ndc, pp, nda .+ 0im) # no method
+=#
```

Comment:
This commented out section probable needs to be deleted?

Reply:
Will do.